torchx/lib/torchx/backend.ex

```diff
@@ -1081,27 +1081,94 @@ defmodule Torchx.Backend do
   end
 
   @impl true
-  def pad(out, tensor, constant, config) do
+  def pad(out, tensor, constant, input_config) do
     config =
-      config
+      input_config
       |> Enum.map(fn {a, b, c} ->
-        if a < 0 or b < 0 or c != 0 do
+        if c < 0 do
           raise ArgumentError, "{#{a}, #{b}, #{c}} padding is not supported"
         end
 
-        [a, b]
+        [max(a, 0), max(b, 0)]
       end)
       |> Enum.reverse()
       |> List.flatten()
 
     constant = Nx.to_number(constant)
 
     tensor
-    |> from_nx()
+    |> pad_internal(input_config)
```

Comment:
Still draft because while this works, the current `pad_internal` implementation might have a better alternative implementation